### PR TITLE
Add missing Krav Maga weapons

### PR DESCRIPTION
**Pull request by ghost**
#### Summary
```SUMMARY: Features "More Krav Maga weapon compatibility"```

#### Purpose of change
The rebalance of Krav Maga in #33602 states the baseline for gun compatibility is automatic rifles and semi-auto pistols. As such, there were service rifles/a service pistol missing from the list.

#### Describe the solution
Added compatibility for the M9, M4A1, M16A4, SCAR-H(L was already allowed), HK416A5 and HK417A2. Also re-alphabetized the SCAR-L given the format of alphabetized melee-pistol-rifle.
#### Describe alternatives you've considered
No love for the Security Guard(M9), Military Recruit(M4A1), or Bionic Soldier(HK416A5). 😢

#### Additional context
Review requested from @Hymore246 / Martial Arts Man.

### CODE REVIEW

File: data/json/martialarts.json [hunk 1]
```diff
@@ -622,18 +623,23 @@
...
+      "hk417_13",
       "hk_g3",
       "hk_g36",
       "l_car_223",
       "l_dsr_223",
       "l_lmg_223",
       "l_mbr_223",
+      "m4a1",
       "m14ebr",
+      "m16a4"
```
Comment (ZhilkinSerg) [inline data/json/martialarts.json]:
```suggestion
      "m16a4",
```

**Comment by tenmillimaster** [comment]
Are there some sources I could see about service weapons being used in krav maga? I'm not finding anything.

**Comment by kevingranade** [comment]
The jist of it is that Krav Maga practitioners in the Israeli armed forces say they train with their service weapons as melee weapons.  I haven't found any really good sources.

**Comment by LaVeyanFiend** [comment]
The source Hymore used was http://www.your-krav-maga-expert.com/krav-maga-weapons.html

> Oh, nevermind, I take it this isn't so much using the service weapon as a weapon as it is fighting while holding a gun?

It is using the service weapon in melee

**Comment by LaVeyanFiend** [comment]
Approval from Hymore obtained.
image

**Comment by tenmillimaster** [comment]
>The jist of it is that Krav Maga practitioners in the Israeli armed forces say they train with their service weapons as melee weapons. I haven't found any really good sources.

So... do Krav Maga teachers in the United States (not in israel) teach how to do this sort of thing?

**Comment by Night-Pryanik** [comment]
Damn, somehow missed this one (and #33602 too). In short, #25598 made it so style name will be shown only to melee weapons, so wielded pistols or rifles won't show style name.

**Comment by Hymore246** [comment]
> 
> 
> Damn, somehow missed this one (and #33602 too). In short, #25598 made it so style name will be shown only to melee weapons, so wielded pistols or rifles won't show style name.

Hm, that's interesting. I always though style names were was automatically applied to weapon descriptions if it was on a whitelist. I'll look into that later.

As for the guns, anything is fine for Krav Maga if it is an automatic pistol or automatic rifle. I did my best with the research but I couldn't find anything about what specific weapons people are trained with in Krav Maga. So I had to take a more general approach to the weapons. CDDA isn't the best when it comes to searching for guns. The whole gun system must have been made by someone who loves guns and as a person without a lot of experience with firearms, I had to go through by hand and look at each pistol and rifle and added them if the description sounded correct. It doesn't surprise me that I missed some. So, good job on finding the rest!

**Comment by tenmillimaster** [comment]
These sources don't really clarify whether they're about disarming people with pistols and rifles. The IDF one does, but again, that's the IDF. In Israel.

If you're using a firearm as a "blunt implement", the type of firearm it is won't matter. The weight and volume of the firearm would matter a lot more. I think that martial arts using any non-long arm would be fine, so long as using it precludes some of the techniques of the art.

I think any martial art using a longarm should be limited to Bayonet Fencing and other such military-only style taught in camps (not krav maga).

Letting an art that uses a firearm very rarely, as it seems in krav maga, sounds like feature creep.

**Comment by Hymore246** [comment]
> Letting an art that uses a firearm very rarely, as it seems in krav maga, sounds like feature creep.
This isn't feature creep. Brawling and Taekwondo already let you fight with firearms.

**Comment by tenmillimaster** [comment]
> > Letting an art that uses a firearm very rarely, as it seems in krav maga, sounds like feature creep.
> 
> This isn't feature creep. Brawling and Taekwondo already let you fight with firearms.

Am I missing something here? This does not appear to be the case, looking at martial_art.json
https://github.com/CleverRaven/Cataclysm-DDA/blob/9e8ad15b899fc11fa403430684c819389c7a58c6/data/json/martialarts.json#L811

https://github.com/CleverRaven/Cataclysm-DDA/blob/9e8ad15b899fc11fa403430684c819389c7a58c6/data/json/martialarts.json#L110

**Comment by Hymore246** [comment]
You can see the tags that let them use any weapon under the **leg_block** attribute: **force_unarmed** and **allow_melee**.

**force_unarmed** - style lets you use the techniques of the style regardless of the equipped weapon. You cannot blocked with the held item.

**allow_melee** - style treats ANY equipped item as if it was on the weapon whitelist. Unlike **force_unarmed**, you can still blocked with the weapon if it lets you.

**force_unarmed** has already been around and is the signature ability of Taekwondo. The logic for **allow_melee** was always there but I added the attribute myself awhile back to allow it to be toggled and to let it not break the logic in martialarts.cpp

**Comment by tenmillimaster** [comment]
Thank you for explaining. I'm a bit more sold on the weapons' use now as bludgeons, but I think this is going back to our 'learn from a book' problem still.

**Comment by LaVeyanFiend** [comment]
There is still the problem of weapon bloat, there are a lot of compatible guns since Imre Lichtenfeld started teaching the style at the IDF. Need another qualifier for style inclusion.

**Comment by LaVeyanFiend** [comment]
Closing until I get more research on what weapons KM primarily uses in the U.S.

**Pull request closed without merging**